src/main/java/org/testng/TestRunner.java

```diff
@@ -1139,7 +1139,10 @@ public void onConfigurationSuccess(ITestResult itr) {
   }
 
   void addMethodInterceptor(IMethodInterceptor methodInterceptor) {
-    m_methodInterceptors.add(methodInterceptor);
+      //avoid to add interceptor twice when the defined listeners implements both ITestListener and IMethodInterceptor.
+      if (!m_methodInterceptors.contains(methodInterceptor)) {
```

Comment:
We could have also just converted this list into a `Set<IMethodInterceptor>`

Reply:
Yes, change list to set can also fix this duplicate adding issue, will I drop this fix?

Reply:
@wubin-ericsson - I didn't get you. Your fix ensures that the first instance is what makes it in and everything else gets discarded. If we use a `Set` then the last interceptor object of a class is what will stay and all the earlier ones get overwritten. I dont think it should matter a lot as to which we use. If we use a `Set` we let the data structure by itself take care of duplicates that's all.

Reply:
Yes, understood, thanks. Then I will keep my original fix.

Reply:
@krmahadevan done, could you please help review again? thanks in advance!